framework/infrastructure/database/mongodb/BaseRepositoryMongo.ts

```diff
@@ -2,7 +2,7 @@ import IRepositoryGeneric from '../IRepositoryGeneric';
 import { Mongoose } from 'mongoose';
 import { APIError } from '../../../crosscutting/exceptions/APIError';
 
-export default abstract class BaseRepositoryMongo<T> implements IRepositoryGeneric<T> {
```

Comment:
Qual a razão de ter removido esta implementação? Todos os repositórios não deveriam vir dele, mesmo que fossem de outros bancos, caso não exista isso não teremos ganho de querer trocar de banco no futuro

Reply:
O problema é que vai ter interface para repositório relacional e não relacional. mais podemos ver isso . E que pensei quando for nao relaciona,  não precisa da transação.

Reply:
Tenho duvidas, o usuário da classe não deveria ter que saber se ela esta utilizando um banco relacional ou não!